HippUnfoldHCP/PostHippUnfoldHCP.sh

```diff
@@ -0,0 +1,300 @@
+#!/bin/bash
+set -eu
+pipedirguessed=0
+if [[ "${HCPPIPEDIR:-}" == "" ]]
+then
+  # pipedirguessed=1
+   #fix this if the script is more than one level below HCPPIPEDIR
+   export HCPPIPEDIR="$(dirname -- "$0")/.."
+fi
+
+source "$HCPPIPEDIR/global/scripts/newopts.shlib" "$@"
+source "$HCPPIPEDIR/global/scripts/debug.shlib" "$@"
+
+
+opts_SetScriptDescription "Make some BIDS structures and run HippUnfold"
+
+opts_AddMandatory '--study-folder' 'StudyFolder' 'path' "folder containing all subjects"
+opts_AddMandatory '--subject' 'Subject' 'subject ID' ""
+opts_AddOptional '--hippunfold-dir' 'PhysicalHippUnfoldDIR' 'path' "location of HippUnfold outputs"
+opts_AddOptional '--atlas-hippunfold-dir' 'AtlasHippUnfoldDIR' 'path' "location of Atlas HippUnfold outputs"
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+    log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+opts_ShowValues
+
+log_Msg "Starting PostHippUnfold pipeline for subject: $Subject"
+
+T1wFolder="$StudyFolder/$Subject/T1w"
+AtlasFolder="$StudyFolder/$Subject/MNINonLinear"
+
+if [ -z ${PhysicalHippUnfoldDIR} ] ; then
+  PhysicalHippUnfoldDIR="${T1wFolder}/HippUnfold"
+fi 
+
+if [ -z ${AtlasHippUnfoldDIR} ] ; then
+  AtlasHippUnfoldDIR="${AtlasFolder}/HippUnfold"
+fi 
+
+function PostHippUnfold {
+HippUnfoldFolderOut=${1}
```

Comment:
I would indent the entire function to make it obvious where it ends.

Reply:
Also, there is only one line in the code (in a loop) that calls this function.  Functions are mainly for when you want to call it from more than one line in the code (such as when you have to change the arguments in a case-specific way).

Reply:
That is in fact why it is a function, there are a bunch of variables that change depending upon which HippUnfold output is being used.  If we standardized on only one kind of HippUnfold inputs (e.g., both T1w and T2w) then I wouldn't need the function.  

As it stands it is most of the script and I didn't want to waste the space.  The function may not be needed in the end if we standardize on only one kind of HippUnfold inputs.

Reply:
There is only one variable which behaves inconsistently between the cases, and you already put that conditional logic *into* the body.  Functions are for when you want to move the inconsistency *outside* the body, by manually calling the function more than once with different explicit patterns (instead of looping).  Then it's like copy-paste and edit, but readable.